Make Roadmap moar exciting!! 🤩

bobcatfish opened the pull request:
# Changes

This commit reorganizes the roadmap so some of the less exciting stuff
is at the end. It also:

* Adds a short list at the top, for maximum excitement
* Adds a goal to get to 1.0 in 2019
* Updates re. the [build
proposal](https://github.com/knative/build/pull/569) which resulted in
  the decision that the Build CRD would remain after all

<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->


<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- ~[ ] Includes [tests](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#principles) (if functionality changed/added)~
- ~[ ] Includes [docs](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#principles) (if user facing)~
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
n/a
```

ImJasonH commented on roadmap-2019.md:
```diff
@@ -76,3 +70,17 @@ Get Pipelines into a reliable release cadence, with trustworthy integration
 itself should dogfood Pipelines for our own CI and CD infrastructure, alongside
 or instead of Prow, by having Prow use Pipelines somehow.
+
+## Migrate to Pipelines
```
I would remove this. Migrating Build to Pipelines could be an item on the Build roadmap, but I don't think it makes sense on Pipelines' roadmap. Whether/when to base Build on Pipelines is not even a settled question at this time, so we shouldn't assume we'll do it just yet.

It's possible we could phrase this as "be stable enough to support Knative Build's implementation" but I think that's already met and exceeded by the 1.0 release section above.

bobcatfish replied:
kk sounds good, will remove!

ImJasonH commented on roadmap-2019.md:
```diff
@@ -76,3 +70,17 @@ Get Pipelines into a reliable release cadence, with trustworthy integration
+invest in these resources and migrate existing `Build` users to these new more
+flexible and powerful resources.
+
+## Don't Break Serving
```
We can remove this too actually, since Serving doesn't currently have a dependency on Pipeline. If it does in the future, we should definitely add it, but I think it could also go into a "reliability" objective along the lines of a 1.0 release.

bobcatfish replied:
kk!

bobcatfish commented:
Updated re. your comments @ImJasonH - but let me know if you think it makes more sense to remove the references I added in the 1.0 section to knative/build and knative/serving, i'm happy to do that also

ImJasonH commented:
This wording SGTM :rocket: 

/lgtm
